### ISSUE TITLE
Internationalize strings

### ISSUE THREAD

**Issue by philschatz** [openstax/rex-web#14]
### Description
Before we start adding in copy text, it is probably worthwhile to add string internationalization.

These 2 were brought up:
- https://github.com/yahoo/react-intl
- https://github.com/i18next/react-i18next

Both seem to support Typescript. yahoo's seems to support attributes, and yahoo's seems to be built on top of the browser i18n APIs. Not sure about the `i18next` one.

#### Branches

Both packages were checked by replacing the `page not found` with internationalized text:

- https://github.com/openstax/rex-web/compare/i18n-react-intl
- https://github.com/openstax/books-web/compare/i18n-react-i18next

#### Notes

The `react-i18next` package seems to offer these features & drawbacks:

- has option to load the translations via xhr but they _can_ be explicitly included (not sure if they can be nested)
- can translate components that have children and can move them to different positions in the translated text (`Hello [Phil], how are you? <avatar/>` vs `Hola! <avatar/> [Phil]`)
- does not seem to error when a **default is not provided**
- does not seem to do the translations for unit tests
- displays the token when something is not found in the translation file

The `react-intl` package seems to offer these features/drawbacks:

- simpler API (no need for an `i18n.js` for configuring
- might be the only one that uses browser i18n APIs
- _might_ support text with elements in it (& reordering those elements). e.g. `Hello [Phil], how are you? <avatar/>` vs `Hola! <avatar/> [Phil]`
- uses `id="i18ntokenhere"` which might be confusing
- unsure what happens when a token is missing
Context: https://openstax.slack.com/archives/C8V1XET1R/p1542216834346800

#### Katalyst notes:

> I was using `http://kazupon.github.io/vue-i18n/` for Vue.js app. It seems like your proposition of `i18next` is very similiar so I can say that I had good experiences with this kind of internationalization. However I wasn't using all of the functions, only basic ones.

> With current internationalization library for webview there was problem, because everything had to be in the same order and if someone left additional `space` or `enter` then it was breaking. With `i18n` this shouldn't be a problem.

Source: https://openstax.slack.com/archives/CBW9DFKGF/p1542314135166300

### Checklist for Done
1A-PRE DESIGN
- [ ] Description is complete 
- [ ] High-level natural language acceptance tests are complete
- [ ] If UX design is needed then the UX label is added
- [ ] If UX design is needed then the UX LoFi Design is completed, approved by the Product Manager, and added to the issue.
- [ ] A high-level solution (development approach for the code change) is added to the issue
- [ ] If the issue is a story, then Epic is selected

1B-DECOMP
- [ ] The work is broken into reasonable and consistent work items, and work item issues created. The work item issues are moved below the Done marker in the Decomp column.
- [ ] Estimate is updated for the work item issues.
- [ ] If there is a parent story epic issue, it is linked with the child work item issues and moved to the Epic column.
- [ ] If an issue is included in the quarterly plan, then Milestone is selected
- [ ] If an issue is included in a product phase, then Release is selected.

1C-DESIGN
- [ ] The UX HiFi Design is complete, approved by the Product Manager, and added to the issue.
- [ ] Distilled acceptance tests are added to the issue.
- [ ] Automated testing criteria are added to issue.
- [ ] The solution is updated.

2A-CODE
  - [ ] A pull request is opened, the automated pull request checks pass, and the change has been approved by other developers.
  - [ ] The acceptance tests are finalized in "Given / When / Then" format.
  - [ ] Regression test categories are identified on the issue. Categories should be referenced if any changes might affect them, even if the intended functionality is unchanged.

2B-PROD ACCEPT
- [ ] The pull request is merged into master by the Product Manager, and the change branch is deleted.

3A-PEND RELEASE
- [ ] All the changes in the release are deployed to the Release Candidate environment.
- [ ] The release is declared frozen by the Product Manager.

3B-RELEASE REVIEW
- [ ] Acceptance tests pass for all issues in the release
- [ ] Blessing from Product Manager that the release is good to go.

3C-RELEASED
- [ ] Production is updated and verified.
- [ ] All issues in the release are closed.

**Comment by kjdav** [comment]
no ux no qa right now
